x-pack/plugins/upgrade_assistant/public/application/components/kibana_deprecations/resolve_deprecation_modal.tsx

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FunctionComponent } from 'react';
+import { i18n } from '@kbn/i18n';
+
+import { EuiConfirmModal } from '@elastic/eui';
+import type { DomainDeprecationDetails } from 'kibana/public';
+
+interface Props {
+  closeModal: () => void;
+  deprecation: DomainDeprecationDetails;
+  isResolvingDeprecation: boolean;
+  resolveDeprecation: (deprecationDetails: DomainDeprecationDetails) => Promise<void>;
+}
+
+const i18nTexts = {
+  getModalTitle: (domainId: string) =>
+    i18n.translate(
+      'xpack.upgradeAssistant.kibanaDeprecations.resolveConfirmationModal.modalTitle',
+      {
+        defaultMessage: "Resolve '{domainId}'?",
```

Comment:
Resolving <plugin name> seems really weird. Maybe Resolve deprecation in <plugin name>? (I don't really like "resolve deprecation" either, but that's how we talk about instances of using deprecated features.